core/file_walker_test.go

```diff
@@ -50,43 +51,62 @@ func createReadDir(ff fakeFile) ReadDir {
 	}
 }
 
+func TestFilePath(t *testing.T) {
+	root := &File{Name: "root"}
+	h1 := &File{Name: "file1", Dir: root}
+	h2 := &File{Name: "file2", Dir: h1}
+	want := "root/file1/file2"
+	if p := h2.Path(); p != want {
+		t.Errorf("unexpected file path, got '%s', want '%s'", p, want)
+	}
+}
+
 func TestGetSubTreeOnSimpleDir(t *testing.T) {
-	testStructure := fakeFile{"a", 0, []fakeFile{
-		fakeFile{"b", 0, []fakeFile{
-			fakeFile{"c", 100, []fakeFile{}},
-			fakeFile{"d", 0, []fakeFile{
-				fakeFile{"e", 50, []fakeFile{}},
-				fakeFile{"f", 30, []fakeFile{}},
-				fakeFile{"g", 70, []fakeFile{ //thisfolder should get ignored
-					fakeFile{"h", 10, []fakeFile{}},
-					fakeFile{"i", 20, []fakeFile{}},
+	testStructure := fakeFile{"a", nil, 0, []fakeFile{
+		fakeFile{"b", nil, 0, []fakeFile{
+			fakeFile{"c", nil, 100, []fakeFile{}},
+			fakeFile{"d", nil, 0, []fakeFile{
+				fakeFile{"e", nil, 50, []fakeFile{}},
+				fakeFile{"f", nil, 30, []fakeFile{}},
+				fakeFile{"g", nil, 70, []fakeFile{ //thisfolder should get ignored
+					fakeFile{"h", nil, 10, []fakeFile{}},
+					fakeFile{"i", nil, 20, []fakeFile{}},
 				}},
 			}},
 		}},
 	}}
 	ignoredFolders := map[string]struct{}{"g": struct{}{}}
-	result := GetSubTree("b", createReadDir(testStructure), ignoredFolders)
-	expected := File{"b", 180, true, []*File{
-		&File{"c", 100, false, []*File{}},
-		&File{"d", 80, true, []*File{
-			&File{"e", 50, false, []*File{}},
-			&File{"f", 30, false, []*File{}},
-		}},
-	}}
-	if !reflect.DeepEqual(*result, expected) {
+	result := GetSubTree("b", nil, createReadDir(testStructure), ignoredFolders)
+	buildExpected := func() *File {
+		b := &File{"b", nil, 180, true, []*File{}}
+		c := &File{"c", b, 100, false, []*File{}}
+		d := &File{"d", b, 80, true, []*File{}}
+		b.Files = []*File{c, d}
+
+		e := &File{"e", nil, 50, false, []*File{}}
+		e.Dir = d
+		f := &File{"f", nil, 30, false, []*File{}}
+		f.Dir = d
+		d.Files = []*File{e, f}
+
+		return b
+	}
+	expected := buildExpected()
+	if !reflect.DeepEqual(*result, *expected) {
 		t.Error("file tree wasn't walked correctly")
-		fmt.Printf("expected: %v", expected)
-		fmt.Printf("result: %v", result)
+		fmt.Printf("expected: %v", *expected)
+		fmt.Printf("result: %v", *result)
 	}
-
 }
 
 func TestGetSubTreeHandlesError(t *testing.T) {
 	failing := func(path string) ([]os.FileInfo, error) {
 		return []os.FileInfo{}, errors.New("Not found")
 	}
-	result := GetSubTree("xyz", failing, map[string]struct{}{})
-	if !reflect.DeepEqual(*result, File{}) {
-		t.Error("GetSubTree didn't return emtpy file on ReadDir failure")
+	name := "xyz"
+	result := GetSubTree(name, nil, failing, map[string]struct{}{})
+	want := File{Name: name}
```

Comment:
You changed the logic here. This was testing that if the `ReadDir` function returns `err` (most likely because the file doesn't exist or we don't have access rights) the `GetSubTree` returns a `File{}` because that's the best it can do.

If `ReadDir` couldn't find `xyz` I don't think it's correct to return `File{Name: "xyz"}` because that file doesn't exist